### ISSUE TITLE
BUG: RuntimeWarning from overflow in `_beta_ppf`

### ISSUE THREAD

**Issue by NixGD** [scipy/scipy#14722]
### Describe your issue.

I'm inconsistently getting a RuntimeWarning from scipy, seemingly only after running a matplotlib plotting function. This isn't really a problem for me, but it is weird and I didn't see it referenced anywhere so I figured I would at least raise an issue.

I don't get the runtime error if I just run `beta.pdf(xs, 1.2, 1.8)` outside the plotting function. In general the error seems to appear inconsistently, and I'm not entirely sure how to characterise the conditions accurately.

This is possibly related to #13364, since [this comment](https://github.com/scipy/scipy/issues/13364#issuecomment-860052441) has the same error from some tests. I am also on an M1 mac, and using the miniforge version of scipy).

### Reproducing Code Example

```python
import matplotlib.pyplot as plt
from scipy.stats import beta
import numpy as np

xs = np.linspace(0, 1, 100)
plt.plot(xs, beta.pdf(xs, 1.2, 1.8))
beta.ppf(0.01, 1.2, 1.8)
```

### Error message

```shell
/opt/homebrew/Caskroom/miniforge/base/envs/smokey/lib/python3.9/site-packages/scipy/stats/_continuous_distns.py:624: RuntimeWarning: overflow encountered in _beta_ppf
  return _boost._beta_ppf(q, a, b)
```

### SciPy/NumPy/Python version information

1.7.0 1.21.2 sys.version_info(major=3, minor=9, micro=6, releaselevel='final', serial=0)

**Comment by rgommers** [comment]
Thank you for the report @NixGD. This issue is a duplicate of gh-14901 now, which has more info - so I'll close this one. We'll try to fix this for `1.7.3` (first release with arm64 macOS wheels).
